### ISSUE TITLE
New endpoint for metadata of all matches?

### ISSUE THREAD

**Issue by kylejrp** [jakejrichards/cod-api#15]
I found this endpoint - I didn't see it listed in b04.ts:
`${BO4_ENDPOINT}/platform/${platform}/gamer/${username}/matches/zombies/start/0/end/0`
It seems to return the metadata for all matches, instead of just the details of 20 within a time range you specify.

🤔 Maybe useful?

**Comment by jakejrichards** [comment]
Yeah I really should implement a metadata instead of only including the detailed endpoints since this is possible. At first I didn't see a reason to use the metadata endpoint, but since this returns **all** match metadata that seems convincing enough to me 😄I'll take a look at this today

**Comment by kylejrp** [comment]
Awesome! Thanks, man.

Do you know if there's an endpoint for viewing the specific match's details? Eg, take the ID from this endpoint and view the details for just that one match.

**Comment by jakejrichards** [comment]
Not that I am aware of, however you could probably treat the match's `timestamp` as its id, its _probably_ unique enough to hit the endpoint like this:

`.../platform/{}/gamer/{}/matches/{}/start/{timestamp}/end/{timestamp + 1}/details`

^ I think this is what you're looking for.

**Comment by jakejrichards** [comment]
It would definitely be a cool feature to add something like `getMatchByTimestamp(timestamp)`
